cpp/src/arrow/array/array_dict_test.cc

```diff
@@ -1117,12 +1117,12 @@ TEST(TestDictionary, Validate) {
   arr = std::make_shared<DictionaryArray>(dict_type, indices, MakeArray(invalid_data));
   ASSERT_RAISES(Invalid, arr->ValidateFull());
 
-  ASSERT_DEATH(
-      {
-        std::shared_ptr<Array> null_dict_arr =
-            std::make_shared<DictionaryArray>(dict_type, indices, nullptr);
-      },
-      "");
+  //   ASSERT_DEATH(
+  //       {
+  //         std::shared_ptr<Array> null_dict_arr =
+  //             std::make_shared<DictionaryArray>(dict_type, indices, nullptr);
+  //       },
+  //       "");
```

Comment:
did you mean to leave this commented?

Reply:
It's part of a single commit I pushed to investigate a CI issue

```
Author:     Felipe Oliveira Carvalho <felipekde@gmail.com>
AuthorDate: Wed Feb 8 10:13:17 2023 -0300
Commit:     Felipe Oliveira Carvalho <felipekde@gmail.com>
CommitDate: Mon Feb 13 13:24:47 2023 -0300

    TEMPORARY: Comment-out ASSERT_DEATH

diff --git a/cpp/src/arrow/array/array_dict_test.cc b/cpp/src/arrow/array/array_dict_test.cc
index bfa732f16..d1e109c90 100644
--- a/cpp/src/arrow/array/array_dict_test.cc
+++ b/cpp/src/arrow/array/array_dict_test.cc
@@ -1117,12 +1117,12 @@ TEST(TestDictionary, Validate) {
   arr = std::make_shared<DictionaryArray>(dict_type, indices, MakeArray(invalid_data));
   ASSERT_RAISES(Invalid, arr->ValidateFull());
 
-  ASSERT_DEATH(
-      {
-        std::shared_ptr<Array> null_dict_arr =
-            std::make_shared<DictionaryArray>(dict_type, indices, nullptr);
-      },
-      "");
+  //   ASSERT_DEATH(
+  //       {
+  //         std::shared_ptr<Array> null_dict_arr =
+  //             std::make_shared<DictionaryArray>(dict_type, indices, nullptr);
+  //       },
+  //       "");
 }
 
```